newsrevenuehub/rev-engine
# DEV-3337: Add core upgrade prompt to sidebar, migrate to TS

Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

This PR looks a bit hefty, but mostly it is me converting things to TypeScript/adding unit tests as I touched them.

- Adds a prompt to the sidebar to upgrade to Core (see Jira for the criteria for this appearing). Closing it should cause the prompt to go away immediately and not show up for the rest of the user session. It uses local storage to remember this.
- Adds an `outlined` variant to our base Button component.
- Adds the LinkButton component from https://github.com/newsrevenuehub/rev-engine/pull/1040

#### Why are we doing this? How does it help us?

More functionality around upgrading; continues TypeScript/Jest migration.

#### How should this be manually tested? Please include detailed step-by-step instructions.

To test the basic workflow:

1. Create a new account, but do not connect Stripe.
2. Confirm that the prompt doesn't appear in the sidebar. Log out of your session and back in to be sure.
3. Connect your account to Stripe.
4. Confirm that now the prompt appears.
5. Confirm that clicking on the Learn More button takes you to the pricing page in a new tab, and that nothing happens in the app itself.
6. Log out of your session and back in and confirm the prompt is still there.
7. Close the prompt.
8. Navigate around the app and confirm the prompt doesn't show up again.
9. Hit reload in your browser and confirm the prompt doesn't show up.
10. Log out of your session and log back in.
11. Confirm that the prompt is back.

To test other types of users not seeing it:

1. Log in as a Hub admin. Confirm the prompt doesn't show up.
2. Log in as a user who is on the Plus plan. Confirm the prompt doesn't show up.
3. If possible, log in as a user on the Core plan. Confirm the prompt doesn't show up.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

We may want to document the prompt.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

I think I'm going to generalize out the logic of "remember this state until logout" into a hook in DEV-3335.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-3337

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No.
